smart_alert.js

```diff
@@ -86,11 +98,40 @@ class SmartAlert {
         this.elm.style.color = textColor;
     }
 
+    pushAlertIntoArray() {
+        alertsObj[this.position].push(this.elm);
+        this.index = alertsObj[this.position].indexOf(this.elm);
+    }
+
+    removeFromArray() {
+        const index = alertsObj[this.position].indexOf(this.elm);
+        alertsObj[this.position].splice(index, 1);
+        this.arrangeAlerts()
+    }
+
+    //moves every alert in the to their respective positions on the array
+    arrangeAlerts() {
+        for (alert of alertsObj[this.position]) {
+            const index = alertsObj[this.position].indexOf(alert);
+            if (this.position.includes("top")) alert.style.top = `${alert.offsetHeight * index}px`;
```

Comment:
Because it is only multiplied by its own height, it overlaps when other alerts are at different heights.
What about storing the total height of the alerts in a variable and using that to specify the display position?

Reply:
Hmm maybe, I'll try it in a bit. I'm not sure if I added the initial welcome alert to the alertsObj, that could also be a reason for the overlapping.

Reply:
Your solution of storing the total height of the alerts into a variable worked very well! I pushed the changes into this branch.